runtime/lua/vim/lsp/structures/location.lua

```diff
@@ -0,0 +1,156 @@
+local highlight = require('vim.highlight')
+local util = require('vim.lsp.util')
+
+local Position = require('vim.lsp.structures.position')
+
+local api = vim.api
+
+local get_first_or_only = function(item)
+  if item == nil or type(item) ~= 'table' then
+    return item
+  end
+
+  if not vim.tbl_islist(item) then
+    return item
+  else
+    return item[1]
+  end
+end
+
+--- Set of functions for working with `Location` objects.
+--- This set of functions also handles working with `LocationLink` for convenience.
+---
+--@ref https://microsoft.github.io/language-server-protocol/specifications/specification-current/#location
+--@ref https://microsoft.github.io/language-server-protocol/specifications/specification-current/#locationLink
+local Location = {}
+
+Location.to_uri_and_range = function(location)
+  local uri = location.targetUri or location.uri
+  local range = location.targetSelectionRange or location.range
+
+  assert(uri and range, string.format("Locations must have `uri` and `range`: %s", vim.inspect(location)))
+  return uri, range
+end
+
+Location.jump = function(location)
+  location = get_first_or_only(location)
+
+  if location == nil then
+    return false
+  end
+
+  local uri, range = Location.to_uri_and_range(location)
+  if uri == nil then return end
```

Comment:
I haven't decided on how I want to do "error handling" kind of things here. So I haven't tightened that up yet. I don't like checking explicitly for `true, false, nil` because it is too easy to not return one of them when you mean to. So I'm considering how it should be done still.